common/types.go

```diff
@@ -470,6 +471,25 @@ func IPV4CIDRToRegex(cidr string) (string, error) {
 	return "^" + regex + `(\/[0-9]?[0-9])?$`, nil
 }
 
+// IsIPv6 returns whether is a IPV6 addresses or not
+func IsIPv6(str string) bool {
```

Comment:
return ip != nil && len(ip) == net.IPv6len

Could you please use `IsIPv6` in NormalizeAddrForURL ?